specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,568 @@
+
+#### System mode
+
+The priority of load paths in system mode follows the definition of ["Storage Directories and Overrides" in the Configuration Files Sepcification], which is also reflected in the following list of paths:
+
+- `/etc/voa/`
+- `/run/voa/`
+- `/usr/local/share/voa/`
+- `/usr/share/voa/`
+
+#### User mode
+
+The priority of load paths in user mode mirrors that of the system mode.
+Refer to the [XDG Base Directory Specification] for details on default values.
+
+- `$XDG_CONFIG_HOME/voa/`
+- the `./voa/` directory in each directory defined in `$XDG_CONFIG_DIRS`
+- `$XDG_RUNTIME_DIR/voa/`
+- `$XDG_DATA_HOME/voa/`
+- the `./voa/` directory in each directory defined in `$XDG_DATA_DIRS`
+
+### Symlinking
+
+Symlinks can be used in the VOA hierarchy to point to files or directories below one of the [load paths].
+Symlinking to files or directories external to the load paths is not supported.
+
+### Masking
+
+Individual _artifact verifiers_ and _trust anchors_ may be masked using a symlink to `/dev/null`, independent of [technology].
+This constitutes the only allowed exception to the rule of not symlinking to files or directories external to the load paths.
+
+These symlinks are only expected to be used in writable and persistent load paths (that is, only below `/etc/voa/` when in system mode, or below`$XDG_DATA_HOME/voa/` when in user mode).
+Masked verifiers are not considered when retrieving from any of the VOA [load paths].
+
+Masking _artifact verifiers_ and _trust anchors_ may be desirable in situations where a verifier is considered compromised, but a revocation for it can not be provided to users in time.
+VOA offers this mechanism as an alternative to revocation.
+
+### Load logic
+
+Load logic depends on the [technology] in use.
+It is the mechanism by which relevant verifier information from all load paths is retrieved, while considering the [os], [purpose], [context] and [technology] request of a calling application.
+
+For some technologies, the first suitable _signature verifier_ found in the list of [load paths] is used, effectively [overriding] files with the same name in directories lower in the list.
+
+However, in other technologies, more complex [merging] logic may apply, that effectively combines information from different representations of one verifier.
+
+Additionally, the concept of [masking] applies for all technologies.
+
+#### Overriding
+
+In technologies that use overriding, verifier files shadow other variants of the same verifier based on filename equality.
+
+For example, `/etc/voa/fedora/packages/default/ssh/buildserver1.pub` overrides `/usr/share/voa/fedora/package/default/ssh/buildserver1.pub`.
+
+Overriding depends on file name equality and directory structure.
+
+For example, `/etc/voa/fedora/packages/default/ssh/buildserver2.pub` does not override `/usr/share/voa/fedora/package/default/ssh/buildserver1.pub`, but instead adds an additional _signature verifier_.
```

Comment:
These are human readable nomenclatures, but what if the fingerprint of the certificate/key is identical? Should the fingerprint be taken into consideration instead? This would also avoid accidental duplicates.